pkg/config/config.go

```diff
@@ -2,148 +2,232 @@ package config
 
 import (
 	"fmt"
-	"github.com/redhat-developer/odo/pkg/util"
 	"io/ioutil"
 	"os"
 	"os/user"
 	"path/filepath"
+	"reflect"
 	"strconv"
 	"strings"
 
+	"github.com/redhat-developer/odo/pkg/util"
+
 	"github.com/ghodss/yaml"
 	"github.com/pkg/errors"
 )
 
 const (
-	configEnvName  = "ODOCONFIG"
-	configFileName = "odo"
+	globalConfigEnvName = "GLOBALODOCONFIG"
+	localConfigEnvName  = "LOCALODOCONFIG"
+	configFileName      = "odo-config.yaml"
 	//DefaultTimeout for openshift server connection check
 	DefaultTimeout = 1
 )
 
+// Info is implemented by configuration managers
+type Info interface {
+	SetConfiguration(parameter string, value string) error
+	GetConfiguration(parameter string) (interface{}, bool)
+	DeleteConfiguration(parameter string) error
+}
+
 // OdoSettings holds all odo specific configurations
 type OdoSettings struct {
 	// Controls if an update notification is shown or not
-	UpdateNotification *bool `json:"updatenotification,omitempty"`
+	UpdateNotification *bool `json:"UpdateNotification,omitempty"`
 	// Holds the prefix part of generated random application name
-	NamePrefix *string `json:"nameprefix,omitempty"`
+	NamePrefix *string `json:"NamePrefix,omitempty"`
 	// Timeout for openshift server connection check
-	Timeout *int `json:"timeout,omitempty"`
+	Timeout *int `json:"Timeout,omitempty"`
+}
+
+// ComponentSettings holds all component related information
+type ComponentSettings struct {
+
+	// The builder image to use
+	ComponentType *string `json:"ComponentType,omitempty"`
 }
 
 // ApplicationInfo holds all important information about one application
 type ApplicationInfo struct {
 	// name of the application
-	Name string `json:"name"`
+	Name string `json:"Name"`
```

Comment:
👍

Reply:
just so the whole file follows the convention of `CamelCase`